### PR TITLE
(plugin) apps::protocols::tcp - add responsetime mode and rename existing one into connectionstatus

### PR DESCRIPTION
**Pull request by UrBnW**
Hi,

This PR adds a proper TCP `responsetime` mode which mimics :
- `check_icmp` binary ;
- `apps::protocols::snmp::mode::responsetime` mode.

The main purpose of this new TCP `responsetime` mode is to be used as a host check, when `check_icmp` can't be used.
Think about for example of a Windows host (with NSClient++) behind a router / firewall.
This new TCP `responsetime` mode will then allow to check the availability of host, computing proper already known `rta` / `rtmin` / `rtmax` / `pl` perfdata.

Current `apps::protocols::tcp::mode::responsetime` mode has then been renamed to a more appropriated `apps::protocols::tcp::mode::connectionstatus` mode.

Many thx 👍

### CODE REVIEW

**Comment by UrBnW** [comment]
```
# centreon_plugins.pl --plugin=apps::protocols::tcp::plugin --mode=response-time --hostname=google.com --port=80  
OK: TCP port 80 rta 9.529ms, lost 0% | 'rta'=9.529ms;;;0; 'rtmax'=10.925ms;;;0; 'rtmin'=8.965ms;;;0; 'pl'=0%;;;0;100
```

File: apps/protocols/tcp/mode/responsetime.pm [hunk 1]
```diff
@@ -99,105 +93,98 @@ sub check_options {
...
+    } else {
         $self->{output}->add_option_msg(short_msg => 'Please set the port option');
         $self->{output}->option_exit();
     }
+    if (defined($self->{option_results}->{timeout}) && $self->{option_results}->{timeout} =~ /(\d+)/) {
+        $self->{option_timeout} = $1;
+    }
+    if (defined($self->{option_results}->{packets}) && $self->{option_results}->{packets} =~ /(\d+)/) {
+        $self->{option_packets} = $1;
+    }
```
Comment (Sims24) [inline apps/protocols/tcp/mode/responsetime.pm:109]: You should use the following syntax to define a default value for options: 

```
        'timeout:s'  => { name => 'timeout' , default => 5},
        'packets:s'  => { name => 'packets' , default => 5},
```

File: apps/protocols/tcp/mode/responsetime.pm [hunk 2]
```diff
@@ -99,105 +93,98 @@ sub check_options {
...
-    }
-    if (!defined($self->{option_results}->{port})) {
+    $self->{option_timeout} = 5;
+    $self->{option_packets} = 5;
+    if (defined($self->{option_results}->{port}) && $self->{option_results}->{port} =~ /(\d+)/) {
+        $self->{option_port} = $1;
+    } else {
         $self->{output}->add_option_msg(short_msg => 'Please set the port option');
         $self->{output}->option_exit();
     }
```
Comment (Sims24) [inline apps/protocols/tcp/mode/responsetime.pm:103]: You can simply check that an option is defined and not empty. You can of course check that's an integer but we don't got this far most of the time ...

File: apps/protocols/tcp/mode/responsetime.pm [hunk 3]
```diff
@@ -99,105 +93,98 @@ sub check_options {
...
+    my $total_time_elapsed = 0;
+    my $max_time_elapsed = 0;
+    my $min_time_elapsed = 0;
+    my $total_packet_lost = 0;
+    for (my $i = 0; $i < $self->{option_packets}; $i++) {
+        my $timing0 = [gettimeofday];
+        my $return = IO::Socket::INET->new(
             PeerAddr => $self->{option_results}->{hostname},
-            PeerPort => $self->{option_results}->{port},
-            Timeout => $self->{option_results}->{timeout},
```
Comment (Sims24) [inline apps/protocols/tcp/mode/responsetime.pm:127]: Better to keep the common way of accessing value option IMHO

**Review by Sims24 — CHANGES_REQUESTED** [review]
Few things that need to be done. 

I agree it's a good idea to have a much detailed tcp ping, nevertheless renaming modes require some changes on the Plugin Pack packaging and it will take some time to merge it and avoid breaking existing setup.

**Comment by UrBnW** [comment]
Many thanks for your review @Sims24, all points addressed in last commit !

**Review by Sims24 — APPROVED** [review]

**Comment by UrBnW** [comment]
Many thanks @Sims24 👍

**Comment by UrBnW** [comment]
Thank you very much @garnier-quentin ;)

**Pull request merged**